components/connectivity-adapter/internal/appregistry/service/converter_test.go

```diff
@@ -0,0 +1,858 @@
+							URL:               "http://oauth.url",
+							ClientID:          "client_id",
+							ClientSecret:      "client_secret",
+							RequestParameters: nil, // TODO not supported
+						},
+					},
+				},
+			},
+			expected: graphql.ApplicationRegisterInput{
+				APIDefinitions: []*graphql.APIDefinitionInput{
+					{
+						Spec: &graphql.APISpecInput{
+							FetchRequest: &graphql.FetchRequestInput{
+								URL: "http://specification.url",
+								Auth: &graphql.AuthInput{
+									Credential: &graphql.CredentialDataInput{
+										Oauth: &graphql.OAuthCredentialDataInput{
+											URL:          "http://oauth.url",
+											ClientID:     "client_id",
+											ClientSecret: "client_secret",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"API specification with request parameters converted to fetch request": {
+			given: model.ServiceDetails{
+				Api: &model.API{
+					SpecificationUrl: "http://specification.url",
+					SpecificationRequestParameters: &model.RequestParameters{
+						QueryParameters: &map[string][]string{
+							"q1": {"a", "b"},
+							"q2": {"c", "d"},
+						},
+						Headers: &map[string][]string{
+							"h1": {"e", "f"},
+							"h2": {"g", "h"},
+						},
+					},
+				},
+			},
+			expected: graphql.ApplicationRegisterInput{
+				APIDefinitions: []*graphql.APIDefinitionInput{
+					{
+						Spec: &graphql.APISpecInput{
+							FetchRequest: &graphql.FetchRequestInput{
+								URL: "http://specification.url",
+								Auth: &graphql.AuthInput{
+									AdditionalQueryParams: &graphql.QueryParams{
+										"q1": {"a", "b"},
+										"q2": {"c", "d"},
+									},
+									AdditionalHeaders: &graphql.HttpHeaders{
+										"h1": {"e", "f"},
+										"h2": {"g", "h"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"event": {
+			given: model.ServiceDetails{
+				Events: &model.Events{
+					Spec: json.RawMessage(`asyncapi: "1.2.0"`),
+				},
+			},
+			expected: graphql.ApplicationRegisterInput{
+				EventDefinitions: []*graphql.EventDefinitionInput{
+					{
+						//TODO what about name
+						Spec: &graphql.EventSpecInput{
+							Data: ptrClob(graphql.CLOB(`asyncapi: "1.2.0"`)),
+						},
+					},
+				},
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			// WHEN
+			actual, err := sut.DetailsToGraphQLInput(tc.given)
+
+			// THEN
+			require.NoError(t, err)
+			assert.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+func TestConversionApplicationExtToServiceDetails(t *testing.T) {
+
+	type testCase struct {
+		given    graphql.ApplicationExt
+		expected model.ServiceDetails
+	}
+	sut := NewConverter()
+
+	for name, tc := range map[string]testCase{
+		"simple attributes": {
+			given: graphql.ApplicationExt{
+				Application: graphql.Application{
+					ID:          "id",
+					Name:        "name",
+					Description: ptrStringOrNilForEmpty("description"),
+					//TODO IntegrationSystemID
```

Comment:
I think we don't need to think about IntegrationSystemID at all, because it's already provided in labels (for filtering purposes).